templates/model_simple.mustache

```diff
@@ -260,6 +260,23 @@ func (o *{{classname}}) Unset{{name}}() {
 
 {{/required}}
 {{/vars}}
+
+// AssertRequired throws an error if a required field is nil
+// It does not check nested models
+func (o *{{{classname}}}) AssertRequired() error {
+{{#hasRequired}}
+{{#requiredVars}}
+{{^nullable}}
+	if IsNil(o.{{name}}) {
+		return MissingRequiredFieldError("{{baseName}}")
```

Comment:
It may be safer to log a warning and decide how to proceed once we understand the impact (Terraform provider conversion would help for this).

Reply:
That would avoid breakage but means the SDK still can't differentiate between types based on required fields (so, for example, the SDK still wouldn't be able to determine that a thing that doesn't have `assigned_to` is not a valid IP assignment).

Reply:
I lost the connection to https://github.com/equinix-labs/metal-go/pull/153#issuecomment-1730203076.

I must have also missed where the generated code iterates through possible oneOf types and uses serialization (and serialization errors) to identify the best-fit type.  That supports the error approach.